apps/api/src/app/partner-integrations/usecases/complete-vercel-integration/complete-vercel-integration.usecase.ts

```diff
@@ -100,25 +91,58 @@ export class CompleteVercelIntegration {
     }
   }
 
+  async findPartnerConfigurationDetail({
+    organizationId,
+    configurationId,
+  }: {
+    organizationId: string;
+    configurationId: string;
+  }) {
+    const organization = await this.organizationRepository.findOne(
+      {
+        _id: organizationId,
+        'partnerConfigurations.configurationId': configurationId,
+      },
+      { 'partnerConfigurations.$': 1 }
+    );
+
+    const configuration = organization?.partnerConfigurations?.find(
```

Comment:
Consider moving this query to Mongo, instead of fetching everything and then do it in JS.

Reply:
the `configurationId` is added to the mongo query, but the thing is that there might be multiple partner configurations, and I'm looking just to get one